Pimp up this thing

msfjarvis opened the pull request:
Stuff :smirk:

msfjarvis committed:
Spectrum: Switch AsyncTask to Thread and variablize props

msfjarvis committed:
Spectrum: Consolidate props

msfjarvis committed:
Spectrum: Make props class and vars package-private

msfjarvis committed:
Spectrum: Nuke LaunchTile

All you really need is the sexy profile tile

msfjarvis committed:
Spectrum: Fork out utility functions into its own class

msfjarvis committed:
Spectrum: Allow kernel developers to remove support for specific profiles

Not all kernel developers ship all 4 profiles Spectrum supports, so add a
simple mechanism for kernel devs to be able to get rid of no-op tiles.

set the spectrum.disabledprofiles prop as a comma separated list of profiles
you DO NOT ship to remove the card for it from the Spectrum UI. Valid profile
names are balance, performance, gaming, and battery.

msfjarvis committed:
Spectrum: Bump to v2.0

msfjarvis committed:
Spectrum: Allow for custom descriptions to be used

This patch allows kernel developers to supply customised descriptions
for each profile. Simply create a /sdcard/.spectrum_descriptions files
following the format below and Spectrum will automagically load them.

```
balance:The best profile for overall smoothness and good battery life.
performance:Max out everything for performance at the cost of battery life and generation of excess heat
gaming:That extra little boost over the performance profile for your graphics heavy games
battery:Tuned to get you that extra hour of SOT you need to last through the day
```
You can also skip certain profiles if you like to retain the stock
descriptions

msfjarvis committed:
Utils: getprop doesn't require root privileges

frap129 commented:
Wow, thanks for all of this. My only question, what's the advantage of having the prop variables in a separate file?

msfjarvis committed:
Spectrum: Make ProfileTile aware of disabled profiles

Also change the profile preference to keep profile names as values
than numbers since it reduces code changes in the ProfileTile. Collapsed
some methods in ProfileTile as well.

msfjarvis committed:
Spectrum: Update gradle

frap129 commented:
Ignore that code review, I'll test when I get home and merge. Still wondering about the props file though

EDIT: Apparently only I can see the code review message

msfjarvis committed:
Spectrum: Respect codacy lint warnings

Get rid of the Props class and move the vars from it into Utils

msfjarvis commented:
@frap129 I've removed the separate Props class and moved the variables from it into the Utils class.

msfjarvis committed:
Utils: More lint fixes from codacy

frap129 commented:
@MSF-Jarvis Testing everything now. Thanks again for this, my hard drive actually died while I was working on some of these things I hadn't got around to restarting them

msfjarvis committed:
Spectrum: Remove untranslatable stubs from de translations

msfjarvis commented:
Rebased the PR on top of the current master

msfjarvis committed:
Utils: Change some vars to local

msfjarvis committed:
Spectrum: Streamline the root and kernel support check workflow

msfjarvis committed:
Spectrum: Optimize the custom description loading

* Only run checks in onCreate if we can access the custom descriptions files
* Reload activity on storage permissions grant to grab the custom descriptions
  and show a toast when permissions are denied. Thanks to @Mazda-- for the onRequestPermissionsResult
  reference

frap129 commented on app/src/main/java/org/frap129/spectrum/MainActivity.java:
```diff
@@ -272,5 +277,26 @@ public boolean onOptionsItemSelected(MenuItem item) {
     }
+    @Override
+    public void onRequestPermissionsResult(int requestCode, @NonNull String permissions[], @NonNull int[] grantResults) {
+        switch (requestCode) {
```
this switch should have a default case

msfjarvis committed:
MainActivity add default case for onRequestPermissionsResult

Commit:
Merge branch 'master' into master

frap129 commented:
Everything seems to work, thanks again!
Edit: Merge conflicts, will fix up when I get home

ghost commented:
Yay

msfjarvis commented:
Why do I even try

frap129 commented:
shit thought I merged it, forgot about this. Sorry man, I'll actually merge it
 this time
